aiida/schedulers/plugins/slurm.py

```diff
@@ -214,6 +214,19 @@ def _get_joblist_command(self, jobs=None, user=None):
                 if not isinstance(jobs, (tuple, list)):
                     raise TypeError("If provided, the 'jobs' variable must be a string or a list of strings")
                 joblist = jobs
+
+            # Trick: If we are asking for a single job, we append the same job once more.
+            # We do this because squeue returns a non-zero exit code if provided an invalid jobid
```

Comment:
```suggestion
            # We do this because we want to have a reliable way of knowing if the squeue command
            # failed - if it didn't fail, we will assume that jobs missing from the output are completed.
            # If, therefore, there is an error that provides an empty output, and we don't detect it, 
            # AiiDA will end up believing that all jobs are completed and will start retrieving them.
            # We therefore want to rely on the exit code of squeue.
            # However, squeue returns a non-zero exit code if provided a *single* invalid jobid
```

Reply:
Thanks Gio.

I merge in your first sentence and try to incorporate the gist of the remaining ones as well.